server/data/restClient.ts

```diff
@@ -78,7 +78,7 @@ export default class RestClient {
         .timeout(this.timeoutConfig())
 
       if (result.notFound) {
-        return undefined
+        return (undefined as unknown) as T
       }
```

Comment:
here is biggest hack. I suspect we need two versions of `get` , `get` and `getOrUndefined`  to get around that `T` is only a union with `undefined` when we explicitly say "ignore 404s".

We can't make the type `T | undefined` since that will cause the majority of cases to deal with `undefined` when it will never return `undefined`

Reply:
That sounds a bit nicer solution anyway.. having two methods